modules/ztp-image-based-upgrade-procedure-cancel.adoc

```diff
@@ -0,0 +1,102 @@
+// Module included in the following assemblies:
+// * edge_computing/image-based-upgrade/ztp-image-based-upgrade.adoc
+
+:_mod-docs-content-type: CONCEPT
+[id="ztp-image-based-upgrade-procedure-cancel_{context}"]
+= Canceling an image-based upgrade on managed clusters at scale
+
+Cancel the upgrade on a set of managed clusters that completed the `Prep` stage.
+
+[IMPORTANT]
+====
+Ensure that the action combinations created by appending actions are supported.
+For more information, see _Supported action combinations_.
+====
+
+.Prerequisites
+
+* Log in to the hub cluster as a user with `cluster-admin` privileges.
+
+.Procedure
+
+. Check the status of the upgrade by running the following command:
++
+[source,terminal]
+----
+$ oc get ibgu -o yaml
+----
+
++
+.Example output
+[source,yaml]
+----
+[...]
+status:
+  clusters:
+  - completedActions:
+    - action: Prep
+    name: spoke1
+  - completedActions:
+    - action: Prep
+    - action: Upgrade
+    - action: FinalizeUpgrade
+    name: spoke4
+  - failedActions:
+    - action: Prep
+    name: spoke6
+[...]
+----
+
+. Create a new file, named `abort-upgrade.yaml`, that contains the action that you want to append to the plan:
++
+[source,yaml]
+----
+  - actions:
+    - Abort
+    rolloutStrategy:
+      maxConcurrency: 5
+      timeout: 10
+----
+
+. Add the `Abort` action to your existing `ImageBasedGroupUpgrade` CR by running the following command:
++
+[source,terminal]
+----
+$ oc patch ibgu example-group-upgrade --type=merge --patch-file abort-upgrade.yaml
+----
```

Comment:
this does not work since merge replaces the whole part of object, so you need the whole plan:

```
spec:
  plan:
    - actions: ["Prep"]
      rolloutStrategy:
        maxConcurrency: 200
        timeout: 1
    - actions: ["Upgrade"]
      rolloutStrategy:
        maxConcurrency: 200
        timeout: 30
    - actions: ["Rollback"]
      rolloutStrategy:
        maxConcurrency: 200
        timeout: 30
```

However we can use json patch to only mention the new part
for example for adding upgrade using patch file:
```
[
    {
        "op": "add",
        "path": "/spec/plan/-",
        "value": {
            "actions": ["Upgrade"],
            "rolloutStrategy": {
                "maxConcurrency": 200,
                "timeout": 30
            }
        }
    }
]
```
```
oc patch ibgu upgrade-4.15.0 --type=json --patch-file samples/patch.json
```
or alternatively in one line:

```
oc patch ibgu upgrade-4.15.0 --type=json -p '[{"op": "add", "path": "/spec/plan/-", "value": {"actions": ["Upgrade"], "rolloutStrategy": {"maxConcurrency": 200, "timeout": 30}}}]'
```